exec/utils.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024-Present Defense Unicorns
+
+package exec
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/defenseunicorns/pkg/helpers"
+)
+
+var registeredCmdMutations = map[string]string{}
+
+// GetFinalExecutablePath returns the absolute path to the current executable, following any symlinks along the way.
+func GetFinalExecutablePath() (string, error) {
+	binaryPath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	// In case the binary is symlinked somewhere else, get the final destination
+	return filepath.EvalSymlinks(binaryPath)
+}
+
+// RegisterCmdMutation registers local ./ commands that should change to the specified cmdLocation
+func RegisterCmdMutation(cmdKey string, cmdLocation string) {
+	registeredCmdMutations[fmt.Sprintf("./%s ", cmdKey)] = fmt.Sprintf("%s ", cmdLocation)
+}
+
+// GetCmdMutation returns the cmdLocation for a given cmdKey and whether that key exists
+func GetCmdMutation(cmdKey string) (string, bool) {
+	cmdLocation, ok := registeredCmdMutations[cmdKey]
+	return cmdLocation, ok
+}
```

Comment:
Note that these functions are not go-routine safe as they edit a map w/ no protections?

Reply:
Do we even want this to be a global registration system?

I think an exported map w/ methods might work better, there is no overlap between how Zarf does its `./zarf` mutation, UDS does its `./uds zarf` mutation, and Maru will do `./run` mutation?

Reply:
This was not intended to be used in go-routines since it is most likely to be configured once within the root command or config code of an app - it was intended to be a way for multiple registrations to occur from different libraries - the thought was that `uds` could register where `./uds` was as well as `./zarf` (and maybe other shorthands like `./kubectl`).  We could move it to be passed into MutateCommand though there isn't an easy way to avoid globals between UDS and Maru because of how they are vendored (and the same would go for any `uds zarf` commands as well)

Reply:
switched to a sync map (though again I wouldn't recommend calling this function from goroutines because race conditions could result in weird outcomes and there shouldn't be a need to edit this map a ton in the first place).